pennylane/ops/op_math/composite.py

```diff
@@ -27,6 +28,24 @@
 # pylint: disable=too-many-instance-attributes
 
 
+def handle_recursion_error(func):
+    """Handles any recursion errors raised from too many levels of nesting."""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except RecursionError as e:
+            raise RuntimeError(
+                "Maximum recursion depth reached! This is likely due to nesting too many "
+                "levels of composite operators. Try setting lazy=False when calling qml.sum "
+                "and qml.prod, or use the + and @ operators instead. Alternatively, you "
+                "can periodically call qml.simplify on your operators."
```

Comment:
Should we include `qml.s_prod` and `*` in this message as well?

Reply:
Oh I forgot about `SProd`. I'll add it.

Reply:
Done